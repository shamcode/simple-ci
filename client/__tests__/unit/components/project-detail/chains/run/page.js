import { createDI } from 'sham-ui';
import Page from '../../../../../../src/components/project-detail/chains/run/page';
import * as directives from 'sham-ui-directives';
import hrefto from 'sham-ui-router/lib/href-to';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const DI = createDI();
    DI.bind( 'store', {
        getProjectChainById: jest.fn().mockReturnValue( Promise.resolve( { project: { id: 1 } } ) ),
        getProjectById: jest.fn().mockReturnValue( Promise.resolve( { project: { id: 1 } } ) )
    } );
    DI.bind( 'router', {
        storage: { params: { id: 1 } },
        generate: jest.fn().mockReturnValueOnce( '/' )
    } );
    const meta = renderer( Page, {
        DI,
        directives: {
            ...directives,
            hrefto
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
