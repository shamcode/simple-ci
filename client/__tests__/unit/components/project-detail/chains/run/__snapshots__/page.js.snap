// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Constructor": "ChainRunPage",
  "Options": Object {
    "ID": "component",
    "chain": Object {},
    "commandOutput": "",
    "containerSelector": "body",
    "dataLoaded": false,
    "dataSaving": false,
    "directives": Object {
      "EventListener": [Function],
      "disabled": [Function],
      "hrefto": [Function],
      "onclick": [Function],
      "oninput": [Function],
      "onload": [Function],
      "onresize": [Function],
      "onscroll": [Function],
      "onselect": [Function],
      "onsubmit": [Function],
      "ref": [Function],
    },
    "errors": Array [],
    "project": Object {},
    "runProjectChain": [Function],
    "types": Array [],
  },
  "html": "
<ul class=\\"breadcrumb\\">
  <li>
    <a href=\\"/\\"> Projects </a>
  </li>
  <!--if-->
  <li></li>
</ul>
<!--BreadCrumb-->
<!--breadcrumbs-->
<!--if-->
<!--FlushPageDataLoading-->
<!--if-->
<!--PageContent-->
",
}
`;
